### PR TITLE
feat: update workflows, add spellcheck, update very_good_analysis

### PR DESCRIPTION
**Pull request by renancaraujo**
## Description

feat: update workflows, add spellcheck, update very_good_analysis

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: .github/cspell.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      "name": "vgv_forbidden",
+      "path": "https://raw.githubusercontent.com/verygoodopensource/very_good_dictionaries/main/forbidden.txt",
+      "description": "Forbidden VGV Spellings"
+    }
+  ],
+  "useGitignore": true,
+
+  "words": [
+  ]
```
Comment (wolfenrain) [inline .github/cspell.json:20]: Can we omit it if it is empty?

File: .github/cspell.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      "description": "Allowed VGV Spellings"
+    },
+    {
+      "name": "vgv_forbidden",
+      "path": "https://raw.githubusercontent.com/verygoodopensource/very_good_dictionaries/main/forbidden.txt",
+      "description": "Forbidden VGV Spellings"
+    }
+  ],
+  "useGitignore": true,
+
```
Comment (wolfenrain) [inline .github/cspell.json:18]:
```suggestion
```

**Review by felangel — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]

File: .github/cspell.json (hunk 1 above)
Comment (renancaraujo) [reply .github/cspell.json:19]: IDK, I like to have this here so when we add something, we have a tighter diff

Comment (renancaraujo) [reply .github/cspell.json]: nah it works fine if we omit, removed whatever

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**